vignettes/datatable-joins.Rmd

```diff
@@ -0,0 +1,687 @@
+  unit = c("unit","ounce"),
+  type = "processed",
+  tax_prop = c(0.65, 0.20)
+)
+
+NewTax
+```
+
+
+3. Defining a `data.table` simulating the products received every Monday with a `product_id` that is not present in the `Products` table.
+
+```{r}
+set.seed(2156)
+
+ProductReceived = data.table(
+  id = 1:10,
+  date = seq(from = as.IDate("2024-01-08"), length.out = 10L, by = "week"), 
+  product_id = sample(c(NA_integer_, 1:3, 6L), size = 10L, replace = TRUE),
+  count = sample(c(50L, 100L, 150L), size = 10L, replace = TRUE)
+)
+
+ProductReceived
+```
+
+4. Defining a `data.table` to show some sales that can take place on weekdays with another `product_id` that is not present in the `Products` table.
+
+```{r}
+sample_date = function(from, to, size, ...){
+  all_days = seq(from = from, to = to, by = "day")
+  weekdays = all_days[wday(all_days) %in% 2:6]
+  days_sample = sample(weekdays, size, ...)
+  days_sample_desc = sort(days_sample)
+  return(days_sample_desc)
+}
+
+set.seed(5415)
+
+ProductSales = data.table(
+  id = 1:10,
+  date = ProductReceived[, sample_date(min(date), max(date), 10L)], 
+  product_id = sample(c(1:3, 7L), size = 10L, replace = TRUE),
+  count = sample(c(50L, 100L, 150L), size = 10L, replace = TRUE)
+)
+
+
+ProductSales
+```
+
+## 2. `data.table` joining syntax 
+
+Before taking advantage of the `data.table` syntax to perform join operations we need to know which arguments can help us to perform successful joins.
+
+The next diagram shows a description for each basic argument. In the following sections we will show how to use each of them and add more complexity little by little.
+
+```
+x[i, on, nomatch]
+| |  |   |
+| |  |   ----> If NULL only returns rows linked in x and i tables
+| |  ----> a character vector o list defining match logict
+| ----> principal data.table, list or data.frame
+----> secundary data.table
+```
+
+An important difference between the regular `data.table` syntax is that _the only argument you can pass by position is the `i` argument_, the rest as you will see are going to be passed by name, so **feel free to change the argument order any time using the argument names** if it seems more convenient.
+
+## 3. Equi joins
+
+This the most common and simple case as we can find common elements between tables to combine.
+
+The relationship between tables can be:
+
+- **One to one**: When each matching value is unique on each table.
+- **One to many**: When some matching values are repeated in one of the table both unique in the other one.
+- **Many to many**: When the matching values are repeated several times on each table.
+
+In most of the following examples we will perform *one to many* matches, but we are also going to take the time to explain the resources available to perform *many to many* matches.
+
+
+### 3.1. Right join
+
+Use this method if you need to combine columns from 2 tables based on one or more references but ***keeping all rows present in the table located on the right***.
```

Comment:
I think "table in the square brackets" or "`i` argument" is more clar than "table located on the right"

Reply:
> change right to `i` ? or at least mention `i` in addition to right

I think it can be confusing as the left join it's also an `i` join, just exchanging the position of each table